rpc/l1.go

```diff
@@ -0,0 +1,112 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/crypto/sha3"
+)
+
+var ErrL1ClientNotFound = jsonrpc.Err(jsonrpc.InternalError, fmt.Errorf("L1 client not found, cannot serve starknet_getMessage"))
+
+type LogMessageToL2 struct {
+	FromAddress *common.Address
+	ToAddress   *common.Address
+	Nonce       *big.Int
+	Selector    *big.Int
+	Payload     []*big.Int
+	Fee         *big.Int
```

Comment:
why they are not felt-s ?

Reply:
The starknet contract ABI returns uint256s (big.Ints) and addresses (common.Address)
```
{
  "anonymous": false,
  "inputs": [
    {
      "indexed": true,
      "internalType": "address",
      "name": "fromAddress",
      "type": "address"
    },
    {
      "indexed": true,
      "internalType": "uint256",
      "name": "toAddress",
      "type": "uint256"
    },
    {
      "indexed": true,
      "internalType": "uint256",
      "name": "selector",
      "type": "uint256"
    },
    {
      "indexed": false,
      "internalType": "uint256[]",
      "name": "payload",
      "type": "uint256[]"
    },
    {
      "indexed": false,
      "internalType": "uint256",
      "name": "nonce",
      "type": "uint256"
    },
    {
      "indexed": false,
      "internalType": "uint256",
      "name": "fee",
      "type": "uint256"
    }
  ],
  "name": "LogMessageToL2",
  "type": "event"
}
```